src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java

```diff
@@ -154,7 +157,27 @@ public static void checkInvalidPrefixesForTags(String argsString) throws ParseEx
         for (PrefixPosition position : positions) {
             Prefix prefix = position.getPrefix();
             if (!PREFIX_TAG.equals(prefix)) {
-                throw new ParseException("Invalid command prefixes detected. Only allows " + PREFIX_TAG + " prefix.");
+                throw new ParseException(String.format(INVALID_PREFIXES_MESSAGE, PREFIX_TAG));
+            }
+        }
+    }
+
+    /**
+     * Checks for invalid prefixes, i.e., prefixes other than {@code n/}  or {@code d/} in the provided arguments
+     * string.
+     * Throws a ParseException if any prefix other than {@code n/} or {@code d/} is found.
+     *
+     * @param argsString The arguments string to check for invalid prefixes
+     * @throws ParseException If an invalid prefix is found
+     */
+    public static void checkInvalidPrefixesForAddEvent(String argsString) throws ParseException {
+        List<PrefixPosition> positions = findAllPrefixPositions(argsString);
+        List<Prefix> validPrefixes = List.of(PREFIX_NAME, PREFIX_DATETIME);
+
+        for (PrefixPosition position : positions) {
+            Prefix prefix = position.getPrefix();
+            if (!validPrefixes.contains(prefix)) {
+                throw new ParseException(String.format(INVALID_PREFIXES_MESSAGE, validPrefixes));
```

Comment:
mmm, @guohuang88 is planning to change for his `EditEventCommand` as well so we can look into that.

I did consider it initially, but the alternative would probably be for the Command Parser to pass in the list of valid prefixes. I am not sure if this would reduce encapsulation though